dev.py

```diff
@@ -763,6 +797,32 @@ def scipy_tests(cls, args, pytest_args):
                   f"installed at:{mod_path}")
             # runner verbosity - convert bool to int
             verbose = int(args.verbose) + 1
+
+            scipy_code_fetching_gcov_flag = (
+                "import scipy;"
+                "print(scipy.show_config(mode='dicts')['Coverage Information']['GCOV'])")
+            was_built_with_gcov_flag = subprocess.run(["python", "-c",
+                                scipy_code_fetching_gcov_flag.format(lang="'c++'")],
+                                capture_output=True, encoding='utf-8').stdout.strip()
+            if was_built_with_gcov_flag == "true":
+                scipy_code_for_checking_compiler = (
+                    "import scipy;"
+                    "print(scipy.show_config(mode='dicts')['Compilers'][{lang}]['name'])")
+                cpp = subprocess.run(["python", "-c",
+                                      scipy_code_for_checking_compiler.format(lang="'c++'")],
+                                      capture_output=True, encoding='utf-8').stdout.strip()
+                c = subprocess.run(["python", "-c",
+                                    scipy_code_for_checking_compiler.format(lang="'c'")],
+                                    capture_output=True, encoding='utf-8').stdout.strip()
+                fortran = subprocess.run(["python", "-c",
+                                          scipy_code_for_checking_compiler.format(lang="'fortran'")],
+                                          capture_output=True, encoding='utf-8').stdout.strip()
```

Comment:
This whole code block is way too complicated. There is no need for multiple `subprocess.run` calls. `__config__.py` contains the data you want, so you can just import it directly from the build directory (e.g. with `importlib`). As a quick check, do this in `build/scipy` in IPython:
```python
_config = __config__.show(mode='dicts')['Compilers']
_config['c']['name']
```